### PR TITLE
ESM 

### PR DESCRIPTION
**Pull request by flurdy**
Upgrades SSOP to use ESM. 
Upgrade the ESLinter and fix all recommended steps.
Upgrade and configure JSDoc inline with changes.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+ */
+export default {
   rootDir: '.',
   verbose: true,
   resetModules: true,
   clearMocks: true,
   silent: false,
   preset: '@shelf/jest-mongodb',
   testMatch: ['**/src/**/*.test.js'],
+  globalTeardown: '<rootDir>/.jest/teardown.js',
```
Comment (feedmypixel) [inline jest.config.js:12]: Is this working with failing tests ok?

File: package.json [hunk 2]
```diff
@@ -2,27 +2,29 @@
...
+    "lint:types": "tsc --build tsconfig.json",
+    "lint:fix": "eslint --cache --fix .",
+    "lint:js": "eslint --cache .",
     "postversion": "git add package.json package-lock.json && git commit -m $npm_package_version",
     "test": "TZ=UTC jest --coverage --verbose",
     "test:watch": "TZ=UTC jest --watch",
-    "server:watch": "nodemon --exec babel-node ./src",
-    "server:debug": "nodemon --exec babel-node --inspect=9239 ./src",
+    "server:watch": "nodemon --exec tsx --enable-source-maps ./src",
+    "server:debug": "nodemon --exec tsx --enable-source-maps --inspect ./src",
```
Comment (feedmypixel) [inline package.json:27]: please keep `--inspect=9239`

**Review by feedmypixel — COMMENTED** [review]
Good stuff. A few comments
- The `eslint` disable comments, I suspect can be fixed rather than being disabled
- I'd make sure the tests are failing ok. The `jest` global teardown looks like it may prevent failing tests being reported.
- Does this build ok in Docker?

File: .jest/teardown.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default async () => {
+  process.exit(0)
+}
```
[the remark this answers is not in the document]
Comment (flurdy) [reply .jest/teardown.js:3]: If we don't need it I will just remove it

File: jest.config.js (hunk 1 above)
Comment (flurdy) [reply jest.config.js:12]: Tests pass. But I'll just remove it

**Review by feedmypixel — APPROVED** [review]
LGTM 👍

**Pull request merged**